configure.ac

```diff
@@ -1133,6 +1135,10 @@ dnl counter implementations. In 1.63 and later the std::atomic approach is defau
 m4_pattern_allow(DBOOST_AC_USE_STD_ATOMIC) dnl otherwise it's treated like a macro
 BOOST_CPPFLAGS="-DBOOST_SP_USE_STD_ATOMIC -DBOOST_AC_USE_STD_ATOMIC $BOOST_CPPFLAGS"
 
+if test "x$enable_suppress_external_warnings" = "xyes"; then
+  BOOST_CPPFLAGS="`echo $BOOST_CPPFLAGS |sed 's/-I/-idirafter /g'`"
```

Comment:
Why not isystem? It would more closely match the behavior of the -I it replaces.
https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html

Reply:
Because `-idirafter /usr/include` does not break gcc's `#include_next`, so I do not have to explicitly handle `/usr/include` in a different way.

Anyway, the `--enable-isystem` introduced in https://github.com/bitcoin/bitcoin/commit/572db3ce8103940c58cc90f4f770e989cf789835 in https://github.com/bitcoin/bitcoin/pull/14920 is superior to the above.

Reply:
I just rebased on top of https://github.com/bitcoin/bitcoin/commit/572db3ce8103940c58cc90f4f770e989cf789835 and ditched the above piece.

Reply:
Here's a much pared-down version of 572db3c that looks to be effective (applied to Boost only):
https://github.com/bitcoin/bitcoin/compare/master...Empact:2020-02-werror?expand=1

Reply:
https://github.com/bitcoin/bitcoin/compare/master...Empact:2020-02-werror looks good. I confirm it works as expected on FreeBSD 12 with clang 9 - it makes it possible to compile Bitcoin Core with full `-Werror`! \o/

Are you going to create PR from it? If yes, then I will ditch this PR.

nit: the new option's description `enable isystem includes for dependencies and stricter warning checks (default is no)` warrants `s/ and stricter warning checks//`

Thanks!

Reply:
Feel free to cherry-pick and modify the subject. Deferring to you as the original author unless you want to punt.

Reply:
Done. I also picked up the QT changes to get this working also with Clang 10.